contracts/ERC721A.sol

```diff
@@ -90,7 +102,7 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
 
         // Counter overflow is impossible as the loop breaks when uint256 i is equal to another uint256 numMintedSoFar.
         unchecked {
-            for (uint256 i; i < numMintedSoFar; i++) {
+            for (uint256 i = _startTokenId(); i < numMintedSoFar + _startTokenId(); i++) {
```

Comment:
It may be more gas efficient to compare it to a local copy of `_nextTokenId`.

```
uint256 end = _nextTokenId;
...
unchecked {
             for (uint256 i = _startTokenId(); i < end; i++) {
```

Reply:
Here are the results in gas for this change. I can only spot a difference in the gas used during deployment. At least it's saving. But seems only to be the case if another `_startTokenId` than `0` is used. 

I've committed this change. 👍 

# Before with `numMintedSoFar`
<img width="1085" alt="Screenshot 2022-02-10 at 10 00 44" src="https://user-images.githubusercontent.com/839848/153373818-a8e0bafd-ad2c-4557-85d5-71bd4ac7f17e.png">


# After with `end`
<img width="1097" alt="Screenshot 2022-02-10 at 10 02 36" src="https://user-images.githubusercontent.com/839848/153373902-f2d855bc-7935-48ff-bef2-3f68fbf51b39.png">